Adding initial info on linters to the style guide & readme (#1249)
buildkite/docs: styleguide/STYLE.md

`app/views/layouts/_sidebar.html.erb` with a description (for example, `"G Cloud Identity", 'integrations/sso/g-cloud-identity'` ).
> **Note:** Ruby, which keeps the website running, interprets underscores in filenames as hyphens. So if a page is called `octopussy_cat.erb.md`, you need to add it as `octopussy-cat` to the `application.html.erb` file.

### Filenames and filename linting

> Use `snake_case` for filenames. The [`.ls-lint` linter](https://github.com/buildkite/docs/blob/main/.ls-lint.yml) checks if this rule is observed.

Reviewer: ```suggestion
Use `snake_case` for `*.md.erb` files in `pages`. The [`.ls-lint` linter](https://github.com/buildkite/docs/blob/main/.ls-lint.yml) checks if this rule is observed.
```

```
Use the `vale off` syntax before a phrase that needs to be bypassed by the linter and don't forget to turn it on again with `vale on`.

### Markdown linting

> A [markdown linter](https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md) is at work in Buildkite documentation. 

Reviewer: ```suggestion
A [markdown linter](https://github.com/DavidAnson/markdownlint) is at work in Buildkite documentation. 
```

The enabled markdown linting rules are in [`.markdownlint.yaml`](https://github.com/buildkite/docs/blob/main/.markdownlint.yaml) file.

> Please keep the [help](https://github.com/buildkite/docs/tree/main/pages/agent/v3/help) files exempt from markdown linting at all times!

Reviewer: ```suggestion
```

  <h3>Setting agent defaults</h3>
  <p>Use a top-level <code>agents</code> block to <a href="/docs/pipelines/defining-steps#step-defaults">set defaults</a> for all steps in a pipeline.</p>
</section>

> ```

Reviewer: ```suggestion
```

```
